### ISSUE TITLE
commonjs-dynamic-required modules are not transformed by plugin-babel

### ISSUE THREAD

**Issue by JLHwung** [rollup/plugins#1085]
- Rollup Plugin Name: commonjs, babel
- Rollup Plugin Version: 21.0.1, 5.3.0
- Rollup Version: 2.63.0
- Operating System (or Browser): macOS
- Node Version:
- Link to reproduction _(⚠️ read below)_: https://github.com/JLHwung/rollup-commonjs-dynamic-register-not-transformed-by-plugin-babel

### Reproduction steps:
```
pnpm i
pnpm test
```

### Expected Behavior
It should contain the transformed module `module.exports = { a: a };`

### Actual Behavior
It contains `module.exports = { a };`, in other words, the imported module is not transformed by plugin-babel

### Additional Information

The dynamic-required-proxied modules have [a wrapped id starting with `\0`](https://github.com/rollup/plugins/blob/9874740ca685b1600e4319365c21e8c1131e9d95/packages/commonjs/src/helpers.js#L2). After they are loaded by the DynamicRequireProxy, they are not transformed by plugin-babel because [any id containing `\0` is rejected by `pluginutils/createFilter`](https://github.com/rollup/plugins/blob/9874740ca685b1600e4319365c21e8c1131e9d95/packages/pluginutils/src/createFilter.ts#L48).

The dynamic required targets are client files, I believe they should be transformed by plugin-babel and any other transformers. A potential solution is to change how the dynamic required targets id are determined.

This issue blocks https://github.com/babel/babel/issues/14112. In that case, we use `plugin-commonjs` to load https://github.com/mathiasbynens/regenerate-unicode-properties. Each unicode properties script contains `const set = regenerate()` and they not transformed by the `plugin-babel`.

**Comment by shellscape** [comment]
Do we need to come up with more sophisticated means to tag virtual modules, perhaps with metadata?

**Comment by lukastaegert** [comment]
I guess I need to finish #1038 as that will just get rid of all the dynamic register logic.

**Comment by lukastaegert** [comment]
BTW as long as you do not rely on watch mode, the current `@rollup/plugin-commonjs@Beta` should work well and likely solve all issues.

**Comment by JLHwung** [comment]
@lukastaegert Yes. I can confirm `@beta` work for Babel. https://github.com/babel/babel/pull/14124

This issue can be closed once #1038 is merged. Can you update the PR info?
